google/cloud/storage/internal/async/connection_impl.cc

```diff
@@ -115,14 +116,18 @@ future<StatusOr<std::unique_ptr<storage_experimental::AsyncReaderConnection>>>
 AsyncConnectionImpl::ReadObject(ReadObjectParams p) {
   using ReturnType =
       StatusOr<std::unique_ptr<storage_experimental::AsyncReaderConnection>>;
-
   auto current = internal::MakeImmutableOptions(std::move(p.options));
   auto proto = ToProto(p.request.impl_);
   if (!proto) return make_ready_future(ReturnType(std::move(proto).status()));
 
-  auto factory = MakeReaderConnectionFactory(
-      std::move(current), std::move(p.request), *std::move(proto));
-  return factory(storage::Generation(), 0);
+  auto const& resume_policy =
+      current->get<storage_experimental::ResumePolicyOption>();
+  auto connection = std::make_unique<AsyncReaderConnectionResume>(
+      resume_policy(),
+      MakeReaderConnectionFactory(std::move(current), std::move(p.request),
+                                  *std::move(proto)));
```

Comment:
It is a little subtle that `current` remains valid long enough for `resume_policy` to be called.  Perhaps it would be easier all around if we just serialized the materialization of the `std::unique_ptr<ResumePolicy>` before the `MakeReaderConnectionFactory()` call.

Reply:
Done.  I put a comment there, because it think the `()()` is easy to miss too.